Formula/pythran.rb

```diff
@@ -0,0 +1,110 @@
+class Pythran < Formula
+  desc "Ahead of Time compiler for numeric kernels"
+  homepage "https://pythran.readthedocs.io/"
+  url "https://files.pythonhosted.org/packages/2e/e9/bc3c16d683dbba5d34666b9861d1c1fdb07da0539655a3cf23a62e4cafa1/pythran-0.9.11.tar.gz"
+  sha256 "a317f91e2aade9f6550dc3bf40b5caeb45b7e012daf27e2b3e4ad928edb01667"
+  license "BSD-3-Clause"
+  head "https://github.com/serge-sans-paille/pythran.git"
+
+  keg_only <<~EOS
+    this formula is mainly used internally by other formulae.
+    Users are advised to use `pip` to install pythran
+  EOS
+
+  depends_on "gcc" # for OpenMP
+  depends_on "numpy"
+  depends_on "openblas"
+  depends_on "python@3.9"
+  depends_on "six"
+
+  resource "beniget" do
+    url "https://files.pythonhosted.org/packages/50/b9/7d864a50d5d2821686d4374400de45c8e0678e553657343f5f0ea6aff98d/beniget-0.3.0.tar.gz"
+    sha256 "062c893be9cdf87c3144fb15041cce4d81c67107c1591952cd45fdce789a0ff1"
+  end
+
+  resource "decorator" do
+    url "https://files.pythonhosted.org/packages/da/93/84fa12f2dc341f8cf5f022ee09e109961055749df2d0c75c5f98746cfe6c/decorator-4.4.2.tar.gz"
+    sha256 "e3a62f0520172440ca0dcc823749319382e377f37f140a0b99ef45fecb84bfe7"
+  end
+
+  resource "gast" do
+    url "https://files.pythonhosted.org/packages/83/4a/07c7e59cef23fb147454663c3271c21da68ba2ab141427c20548ae5a8a4d/gast-0.4.0.tar.gz"
+    sha256 "40feb7b8b8434785585ab224d1568b857edb18297e5a3047f1ba012bc83b42c1"
+  end
+
+  resource "networkx" do
+    url "https://files.pythonhosted.org/packages/b0/21/adfbf6168631e28577e4af9eb9f26d75fe72b2bb1d33762a5f2c425e6c2a/networkx-2.5.1.tar.gz"
+    sha256 "109cd585cac41297f71103c3c42ac6ef7379f29788eb54cb751be5a663bb235a"
+  end
+
+  resource "ply" do
+    url "https://files.pythonhosted.org/packages/e5/69/882ee5c9d017149285cab114ebeab373308ef0f874fcdac9beb90e0ac4da/ply-3.11.tar.gz"
+    sha256 "00c7c1aaa88358b9c765b6d3000c6eec0ba42abca5351b095321aef446081da3"
+  end
+
+  def install
+    on_macos do
+      gcc_major_ver = Formula["gcc"].any_installed_version.major
+      inreplace "pythran/pythran-darwin.cfg" do |s|
+        s.gsub!(/^include_dirs=/, "include_dirs=#{Formula["openblas"].opt_include}")
+        s.gsub!(/^library_dirs=/, "library_dirs=#{Formula["openblas"].opt_lib}")
+        s.gsub!(/^blas=.*/, "blas=openblas")
+        s.gsub!(/^CC=.*/, "CC=#{Formula["gcc"].opt_bin}/gcc-#{gcc_major_ver}")
+        s.gsub!(/^CXX=.*/, "CXX=#{Formula["gcc"].opt_bin}/g++-#{gcc_major_ver}")
```

Comment:
This sets a default compiler with OpenMP support (`gcc`).
However, it will require a revision bump on `gcc` major version updates.
Also, in formulae, the default `CC`/`CXX` env vars will override this.

Reply:
Will this fail the test if you try to run it without `gcc` installed, or have an outdated one, so it can't find the specific version of `gcc` referenced?

Reply:
It does fail in the 2nd testcase with default `clang` disabled via `with_env(CC: nil, CXX: nil)`.

In the 1st testcase, it will end up using default `clang` and pass.

---

As mentioned though, it seems to be a bit difficult to control the compiler when `pythran` is used as a dependency in another formula as the environment variables override the compiler.

Particularly in the case of something like `scipy` where the usage is embedded inside the code (though `scipy` still passes even when using Apple Clang, so it probably doesn't need `pythran+openmp` feature).

I was thinking that if we do make this formula keg-only specifically as dependency for other formulae, then the impact of Homebrew `ENV.cc` and `ENV.cxx` may make these configs useless and just add maintenance burden.

The GNU `gcc` defaults are really more useful for users that use `pythran` directly on terminal as `-fopenmp` can help produce better results.

Reply:
Seems fine to keep it, I think. The `gcc` dep is free given the `openblas` dep, so may as well make it work as well as possible given that.